### ISSUE TITLE
Build error with ldc-1.18.0

### ISSUE THREAD

**Issue by shabunin** [deviator/serialport#9]
First, thanks for this great library.

Now, I can't compile it with ldc-1.18.0. The output is following:

```text
/home/bobalus/dlang/ldc-1.18.0/bin/../import/std/traits.d(8387,5): Error: template instance isAggregateType!(exception) does not match template declaration isAggregateType(T)
../../.dub/packages/serialport-2.2.0/serialport/source/serialport/exception.d(148,26):        while looking for match for getSymbolsByUDA!(exception, preallocated)
/home/bobalus/dlang/ldc-1.18.0/bin/ldc2 failed with exit code 1.
```

Build is successful with ldc-1.17.0, 1.15.0.

UPD: little investigation reveals that it is troubles with phobos library. 

https://github.com/dlang/phobos/commit/86733d5a4a5379219f04524bead35eff33c6a1a7

Then issue was opened 
https://issues.dlang.org/show_bug.cgi?id=20054

There was another pull request to revert changes, but still is not approved.

**Comment by deviator** [comment]
I add new version `2.2.1`... I hope it will be temporary fix, and in future versions of library I'll can remove this ugly copy-paste from old phobos.

**Comment by deviator** [comment]
oh, new version builds only with ldc 1.17 and 1.18, need new fixes

**Comment by deviator** [comment]
I think it's closed by v2.2.2
